var/spack/repos/builtin/packages/iwyu/package.py

```diff
@@ -33,14 +33,14 @@ class Iwyu(CMakePackage):
     depends_on('llvm+clang@7.0:7', when='@0.11')
 
     # Non-X86 CPU use all_targets variants because iwyu use X86AsmParser
-    depends_on('llvm+all_targets', when='target=aarch64:')
-    depends_on('llvm+all_targets', when='target=arm:')
-    depends_on('llvm+all_targets', when='target=ppc:')
-    depends_on('llvm+all_targets', when='target=ppcle:')
-    depends_on('llvm+all_targets', when='target=ppc64:')
-    depends_on('llvm+all_targets', when='target=ppc64le:')
-    depends_on('llvm+all_targets', when='target=sparc:')
-    depends_on('llvm+all_targets', when='target=sparc64:')
+    depends_on('llvm', when='target=aarch64:')
+    depends_on('llvm', when='target=arm:')
+    depends_on('llvm', when='target=ppc:')
+    depends_on('llvm', when='target=ppcle:')
+    depends_on('llvm', when='target=ppc64:')
+    depends_on('llvm', when='target=ppc64le:')
+    depends_on('llvm', when='target=sparc:')
+    depends_on('llvm', when='target=sparc64:')
```

Comment:
I think these all need to be `targets=x86` since IWYU needs to build x86-related code on non-x86 platforms.

Reply:
@haampie This assumes the llvm dependency is `all` or `x86`, but for correctness shouldn't it include that explicitly?